persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SampleSparkRepository.java

```diff
@@ -0,0 +1,197 @@
+        
+        if (!CollectionUtils.isEmpty(sampleIds)) {
+            samples = samples.where(samples.col("SAMPLE_ID").isin(sampleIds.toArray()));
+        }
+        
+        samples.createOrReplaceTempView("samples");
+        StringBuilder sb = new StringBuilder("SELECT PATIENT_ID, SAMPLE_ID, studyId");
+        if (PersistenceConstants.SUMMARY_PROJECTION.equalsIgnoreCase(projection)
+            || PersistenceConstants.DETAILED_PROJECTION.equalsIgnoreCase(projection)) {
+            sb.append(", SAMPLE_TYPE");
+        }
+        sb.append(" FROM samples");
+
+        Dataset<Row> res = spark.sql(sb.toString());
+        
+        return res.collectAsList().stream().
+            map(r -> mapToSample(r, projection)).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Sample> fetchSamples(List<String> sampleListIds, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaSamples(List<String> studyIds, List<String> sampleIds) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaSamples(List<String> sampleListIds) {
+        throw new UnsupportedOperationException();    }
+
+    @Override
+    public List<Sample> getSamplesByInternalIds(List<Integer> internalIds) {
+        throw new UnsupportedOperationException();
+    }
+
+    private Dataset<Row> loadStudyFiles(Set<String> studyIds, String file) {
+        List<String> studyIdArr = studyIds.stream()
+            .map(s -> PARQUET_DIR + ParquetConstants.STUDIES_DIR + s + "/" + file).collect(Collectors.toList());
+        Seq<String> fileSeq = JavaConverters.asScalaBuffer(studyIdArr).toSeq();
+
+        UserDefinedFunction getStudyId = udf((String fullPath) -> {
+            String[] paths = fullPath.split("/");
+            return paths[paths.length-3];
+        }, DataTypes.StringType);
+
+        return spark.read()
+            .option("mergeSchema", true)
+            .parquet(fileSeq)
+            .withColumn("studyId", getStudyId.apply(input_file_name()));
+    }
+    
+    private Sample mapToSample(Row row, String projection) {
+        Sample sample = new Sample();
+        sample.setPatientStableId(row.getString(0));
+        sample.setStableId(row.getString(1));
+        String typeOrStudy = row.getString(2);
+        if (PersistenceConstants.SUMMARY_PROJECTION.equalsIgnoreCase(projection)
+            || PersistenceConstants.DETAILED_PROJECTION.equalsIgnoreCase(projection)) {
+            Sample.SampleType sampleType = getType(sample.getStableId(), typeOrStudy);
+            sample.setSampleType(sampleType);
+            sample.setCancerStudyIdentifier(row.getString(3));
+        } else {
+            sample.setCancerStudyIdentifier(typeOrStudy);
+
+        }
+        return sample;
+    }
+
+    private Sample.SampleType getType(String stableId, String sampleType) {
```

Comment:
This should be in a utils class and needs to be tested

Reply:
The logic was based on from org.mskcc.cbio.portal.model.Sample.getType

Reply:
Added a util class and more JUnit test cases.